### ISSUE TITLE
Self-service reveal on what all "consumed" a resource "scope"

### ISSUE THREAD

**Issue by BigLep** [libp2p/go-libp2p-resource-manager#25]
## Done Criteria
When a resource limit is exceeded, it's easily determinable what has consumed the allotted resources that are now being consumed.  Said another way, when a resource limit is exceeded, we want to know what was contributing to this, not just the specific call that exceeded the limit.

Ideally this should be in an easily parsable log message.  

It can determined during implementation time whether this full dump of the resource manager state is done by default or needs to be enabled by an environment variable.

## Why Important
We need to make it more self-service for go-libp2p consumers to track down why they are having resource leaks or too-low-of-limits so they can help themselves, not rely on go-libp2p maintainers, and ultimately not default to losing the protections of the resource manager by disabling.   
 
## User/Customer
Developers of applications that consume go-libp2p

## Notes
Currently we output a message like:

```
2022-05-25T13:56:29.298Z	DEBUG	rcmgr	go-libp2p-resource-manager@v0.2.1/scope.go:242	blocked memory reservation	
{
   "scope":"peer:12D3KooWFZy8PomPodoEGem4mKfGhENJTEcLfeXSk2DFSJR8zRnD",
   "size":262144,
   "priority":255,
   "stat":{
      "NumStreamsInbound":2,
      "NumStreamsOutbound":255,
      "NumConnsInbound":1,
      "NumConnsOutbound":0,
      "NumFD":1,
      "Memory":67108864
   },
   "error":"resource limit exceeded"
}
```

This tells us there was a "blocked memory reservation for scope ""peer:12D3KooWFZy8PomPodoEGem4mKfGhENJTEcLfeXSk2DFSJR8zRnD", but it doesn't give clarity on what was taking up the rest of the memory which is causing this reservation to fail.  

Basically, beyond focusing on the "water" that overfiowed the bathtub, we also want to know the story of the water that filled up the bathtub to begin with.

**Comment by marten-seemann** [comment]
Indeed. All that we can do is flag the allocation that caused the bathtub to overflow, but not what filled up the bathtub in the first place. I believe that this is better addressed by monitoring a Grafana dashboard, which gives a much more comprehensive overview of resource usage.
